File: plate-fe/src/app/core/services/equipement.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Equipement } from '../models/equipement.model';

@Injectable({
  providedIn: 'root'
})
export class EquipementService {
  private apiUrl = 'http://localhost:8080/api/equipements';

  constructor(private http: HttpClient) {}

  getAll(): Observable<Equipement[]> {
    return this.http.get<Equipement[]>(this.apiUrl);
  }

  getById(id: string): Observable<Equipement> {
    return this.http.get<Equipement>(`${this.apiUrl}/${id}`);
  }

  getBySalle(idSalle: string): Observable<Equipement[]> {
    return this.http.get<Equipement[]>(`${this.apiUrl}/salle/${idSalle}`);
  }

  create(equipement: Equipement): Observable<Equipement> {
    return this.http.post<Equipement>(this.apiUrl, equipement);
  }

  update(id: string, equipement: Equipement): Observable<Equipement> {
    return this.http.put<Equipement>(`${this.apiUrl}/${id}`, equipement);
  }

  updateEtat(id: string, etat: string): Observable<Equipement> {
    return this.http.patch<Equipement>(`${this.apiUrl}/${id}/etat?etat=${etat}`, {});
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: plate-be/src/main/java/tn/enicarthage/plate_be/services/EquipementService.java
package tn.enicarthage.plate_be.services;

import tn.enicarthage.plate_be.dtos.EquipementDTO;
import tn.enicarthage.plate_be.dtos.EquipementRequestDTO;
import tn.enicarthage.plate_be.entities.ETAT_EQUIPEMENT;

import java.util.List;

public interface EquipementService {

    List<EquipementDTO> getAll();

    EquipementDTO getById(String id);

    List<EquipementDTO> getBySalle(String idSalle);

    EquipementDTO create(EquipementRequestDTO dto);

    EquipementDTO update(String id, EquipementRequestDTO dto);

    EquipementDTO updateEtat(String id, ETAT_EQUIPEMENT etat);

    void delete(String id);

    List<EquipementDTO> getGarantiesExpirees();
}
